server/tls.go

```diff
@@ -0,0 +1,81 @@
+package server
```

Comment:
`tls` works for me. Having a separate package for this makes sense. Lets do it and get things done. When we realize it is a problem later we can fix it. I agree that the server package should not be imported in the storage package.